src/v2/features/characters/characters-value.service.ts

```diff
@@ -0,0 +1,474 @@
+import * as fs from 'fs';
+
+import { ICharacter2 } from 'src/models/interfaces';
+import { Rank, Rarity, RarityStars } from 'src/models/enums';
+
+const readDataFromFile = (filename: string): any => {
+    const rawData = fs.readFileSync(filename);
+    return JSON.parse(rawData.toString());
+};
+
+const rankUpData = readDataFromFile('assets/rankUpData.json');
+const recipeData = readDataFromFile('assets/recipeData.json');
+
+export class CharactersPowerService {
+    public static getCharacterPower(character: ICharacter2): number {
+        if (character.rank === Rank.Locked) {
+            return 0;
+        }
+
+        const powerLevel =
+            CharactersPowerService.getUnlockValue(character.initialRarity, character.name) +
+            CharactersPowerService.getExperienceValue(character.level) +
+            CharactersPowerService.getAbilityValue(character.activeAbilityLevel) +
+            CharactersPowerService.getAbilityValue(character.passiveAbilityLevel) +
+            CharactersPowerService.getStarsValue(character.stars) -
+            CharactersPowerService.getInitialStarsValue(character.initialRarity) +
+            CharactersPowerService.getRarityValue(character.rarity) -
+            CharactersPowerService.getRarityValue(character.initialRarity) +
+            CharactersPowerService.getRankValue(character.rank, character.name);
+        return Math.round(powerLevel);
+    }
+
+    public static getRankValue(currentRank: Rank, character: string): number {
+        const CoinBS: number = 780 / 60000;
+        const MaterialBS: { [key: string]: number } = {
+            Common: 5,
+            Uncommon: 10,
+            Rare: 25,
+            Epic: 60,
+            Legendary: 150,
+        };
+
+        let grandtotal: number = 0;
+
+        // "Rank.indexOf" doesn't work here.
+        for (let thisrank = 0; thisrank < Rank.indexOf(currentRank); thisrank++) {
```

Comment:
There are similar code to determine Ranks range in the Static data service
![image](https://github.com/svehera/tacticusplanner/assets/33064347/c657d292-eea8-4e3b-8749-1cd2be4331a0)

Reply:
I think this is how it should look like
  ```
  const rankEntries: number[] = getEnumValues(Rank).filter(x => x > 0);
        for (let thisrank = 0; thisrank < rankEntries.indexOf(currentRank); thisrank++) {
  ```